### ISSUE TITLE
pipe with isTransformEnabled = false does not revert to plain causing validations to fail

### ISSUE THREAD

**Issue by alexmalev** [nestjs/nest#11070]
### Did you read the migration guide?

- [X] I have read the whole migration guide

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Potential Commit/PR that introduced the regression

https://github.com/nestjs/nest/pull/11028

### NestJS version

9.1.2 -> 9.3.3

### Describe the regression

because of the new default ` this.validatorOptions = { forbidUnknownValues: false, ...validatorOptions };`
validation pipes that should not transform fail to revert their return value to plain and instead return the transformed instance. when the same instance than again goes through a pipe that should transform, it fails because the value has already been transformed.
this is where it fails to revert
```ts
return Object.keys(this.validatorOptions).length > 0
      ? classTransformer.classToPlain(entity, this.transformOptions)
      : value;
```
it used to return the value... now because of the default `forbidUnknownValues` it returns class to plain

### Input code

```ts
// main.ts
app.useGlobalPipes(new ValidationPipe())

//controller
@Get('something')
  async getSomething(
    @Query(new ValidationPipe({ transform: true }))
    { params = [] }: QueryInputDto,
  ): Promise<SomeDto> {
    return something;
  }

//DTO

export default class QueryInputDto {
  @ApiProperty({
    isArray: true,
  })
  @IsArray()
  @IsOptional()
  @IsEnum(SomeEnum, { each: true })
  @Transform(({ value }) => value.split(','))
  params: SomeEnum[];
}
```

### Expected behavior

the @transform should be only called once like it used to before the regression.
it is now called twice. once for the global pipe turning the value into an array. (and does not revert it)

the second time it tries to transofrm the value - but fails since it cant split the array

**Comment by kamilmysliwiec** [comment]
Fixed in 9.3.5

**Comment by alexmalev** [comment]
> Fixed in 9.3.5

That was blazing fast. Thanks
